pkg/store/storepb/custom.go

```diff
@@ -456,3 +461,114 @@ func CompareLabels(a, b []Label) int {
 func LabelsToPromLabelsUnsafe(lset []Label) labels.Labels {
 	return labelpb.ZLabelsToPromLabels(lset)
 }
+
+// Type alias because protoc-go-inject-field does not support
+// managing imports.
+type syncPool = sync.Pool
+
+// The following were copied/pasted from gogoprotobuf generated code with changes
+// to make it work with sync.Pool / []byte slice.
+func (m *SeriesResponse) Marshal() (dAtA []byte, err error) {
+	size := m.Size()
+
+	var respBuf []byte
+	if m.respBuf == nil {
+		respBuf = make([]byte, size)
+	} else {
+		if cap(**m.respBuf) < size {
+			if m.respPool != nil {
+				m.respPool.Put(*m.respBuf)
```

Comment:
Does it mean that pooled buffers will have only 32 bytes only? I think this might limit usability of this pooling. I wonder if we don't want to resize for bigger ones and put resized ones with some maximum length on it.

Reply:
It means that it is the initial size so they would be useful for responses with minimal labels such as `a="b"` and one sample, for example. If there are more samples and/or more labels then they would be resized to be big just enough. I'm not sure I get the suggestion about the maximum length - we don't make them any bigger than they are right now in this PR.